data_steward/cdr_cleaner/clean_cdr_engine.py

```diff
@@ -37,129 +41,225 @@ def add_console_logging(add_handler):
         logging.getLogger('').addHandler(handler)
 
 
-def clean_dataset(project=None, statements=None, data_stage=stage.UNSPECIFIED):
+def create_sandbox(project_id, dataset_id, client):
+    sandbox_dataset_id = sandbox.get_sandbox_dataset_id(dataset_id)
+    labels = {'type': 'sandbox', 'dataset_id': dataset_id}
+    sandbox_dataset = bq.define_dataset(project_id,
+                                        sandbox_dataset_id,
+                                        description=f'Sandbox for {dataset_id}',
+                                        label_or_tag=labels)
+    sandbox_dataset = client.create_dataset(sandbox_dataset, exists_ok=True)
+    return sandbox_dataset.dataset_id
+
+
+def clean_dataset(project_id=None, dataset_id=None, rules=None, **kwargs):
```

Comment:
I think we should like `project_id`, `dataset_id`, `rules` to be required rather than optional arguments.
1. Currently if `rules` is not provided this function will raise an error when it gets to the for loop. This is avoided if the argument is required.
1. The caller may simply forget to set one of these args and inadvertently edit resources (e.g. env var for project is set to prod instead of test).
1. The default values make calls to this function dependent on external configuration (env variables) which can be ambiguous and difficult to manage. If it is a required arg then all the configs are explicit and in one place.
1. There do not appear to be any calls to `clean_dataset()` where the defaults are used. This could be because there has not been a need. I would advise adding defaults to parameters like these only when there is a very important use case.